File: androidclog/src/main/java/com/caseyjbrooks/androidclog/AndroidLogger.java
package com.caseyjbrooks.androidclog;

import android.util.Log;
import com.caseyjbrooks.clog.Clog;
import com.caseyjbrooks.clog.ClogLogger;

public class AndroidLogger implements ClogLogger {

    private Clog.Priority priority;

    public AndroidLogger() {
        this(Clog.Priority.DEFAULT);
    }

    public AndroidLogger(Clog.Priority priority) {
        this.priority = priority;
    }

    @Override
    public boolean isActive() {
        return true;
    }

    @Override
    public int log(String tag, String message) {
        switch (priority) {
            case VERBOSE: return Log.v(tag,   message);
            case DEBUG:   return Log.d(tag,   message);
            case INFO:    return Log.i(tag,   message);
            case WARNING: return Log.w(tag,   message);
            case ERROR:   return Log.e(tag,   message);
            case FATAL:   return Log.wtf(tag, message);
            case DEFAULT:
            default:      return Log.i(tag,   message);
        }
    }

    @Override
    public int log(String tag, String message, Throwable throwable) {
        switch (priority) {
            case VERBOSE: return Log.v(tag,   message, throwable);
            case DEBUG:   return Log.d(tag,   message, throwable);
            case INFO:    return Log.i(tag,   message, throwable);
            case WARNING: return Log.w(tag,   message, throwable);
            case ERROR:   return Log.e(tag,   message, throwable);
            case FATAL:   return Log.wtf(tag, message, throwable);
            case DEFAULT:
            default:      return Log.i(tag,   message, throwable);
        }
    }

    @Override
    public Clog.Priority priority() {
        return priority;
    }

}

File: core/src/androidMain/kotlin/clog/impl/AndroidLogger.kt
package clog.impl

import android.util.Log
import clog.Clog
import clog.api.ClogLogger

class AndroidLogger : ClogLogger {
    override fun log(priority: Clog.Priority, tag: String?, message: String) {
        when (priority) {
            Clog.Priority.VERBOSE -> Log.v(tag, message)
            Clog.Priority.DEBUG -> Log.d(tag, message)
            Clog.Priority.INFO -> Log.i(tag, message)
            Clog.Priority.DEFAULT -> Log.i(tag, message)
            Clog.Priority.WARNING -> Log.w(tag, message)
            Clog.Priority.ERROR -> Log.e(tag, message)
            Clog.Priority.FATAL -> Log.wtf(tag, message)
        }
    }
}
